ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -39,7 +51,14 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
             IQueryable<Notification> draftsIQ = service.GetBaseQueryableNotificationByStatus(draftStatusList);
             IQueryable<Notification> nonDraftsIQ = service.GetBaseQueryableNotificationByStatus(nonDraftStatusList);
 
-            var notificationsIQ = OrderQueryable(draftsIQ).Union(OrderQueryable(nonDraftsIQ));
+            if (!String.IsNullOrEmpty(IdFilter))

```

Comment:
I think a good chunk of this logic could be pushed into the notification service, which would make it easier to get decent unit testing coverage of this logic.